ui/js/graphs.js

```diff
@@ -599,9 +617,15 @@ perf.controller('GraphsCtrl', [
 
           // we only store the signature + project name in the url, we need to
           // fetch everything else from the server
-          var partialSeriesList = $stateParams.series.map(function(encodedSeries) {
-            return JSON.parse(decodeURIComponent(encodedSeries));
-          });
+          
+          var partialSeriesString = decodeURIComponent($stateParams.series.pop());
+          partialSeriesString = partialSeriesString.replace(/[\[\]"]/g, ''); 
+          var partialSeriesArray = partialSeriesString.split(",");
+          var partialSeriesList = Object.create(null)
```

Comment:
I don't think this will work (see below), but that aside, here's a more concise way of doing the same thing:

    var partialSeries = {
      project: partialSeriesArray[0],
      signature: partialSeriesArray[1],
      visible: partialSeriesArray[2]
    }